### ISSUE TITLE
Panic when trying to read non-existent vault secret

### ISSUE THREAD

**Issue by jmccann** [drone/drone#2193]
If you try to reference a vault secret that does not exist in vault, drone will then panic and put the build in a forever pending state.

This is with `drone/drone:0.8.0`.

The stack trace:

```
Sep  5 21:46:11 drone6-server drone6[6307]: runtime error: invalid memory address or nil pointer dereference
Sep  5 21:46:11 drone6-server drone6[6307]: /usr/local/go/src/runtime/panic.go:489 (0x42f29f)
Sep  5 21:46:11 drone6-server drone6[6307]: /usr/local/go/src/runtime/panic.go:63 (0x42e14e)
Sep  5 21:46:11 drone6-server drone6[6307]: /usr/local/go/src/runtime/signal_unix.go:290 (0x44430f)
Sep  5 21:46:11 drone6-server drone6[6307]: /go/src/github.com/drone/drone/extras/secrets/vault.go:93 (0xa160c6)
Sep  5 21:46:11 drone6-server drone6[6307]: /go/src/github.com/drone/drone/extras/secrets/vault.go:52 (0xa15c53)
Sep  5 21:46:12 drone6-server drone6[6307]: /go/src/github.com/drone/drone/server/hook.go:222 (0x96871e)
Sep  5 21:46:12 drone6-server drone6[6307]: /go/src/github.com/drone/drone/vendor/github.com/gin-gonic/gin/context.go:97 (0x8baa1a)
Sep  5 21:46:12 drone6-server drone6[6307]: /go/src/github.com/drone/drone/router/middleware/token/token.go:17 (0x8d3ff0)
Sep  5 21:46:12 drone6-server drone6[6307]: /go/src/github.com/drone/drone/vendor/github.com/gin-gonic/gin/context.go:97 (0x8baa1a)
Sep  5 21:46:12 drone6-server drone6[6307]: /go/src/github.com/drone/drone/router/middleware/session/user.go:68 (0x8d3492)
Sep  5 21:46:12 drone6-server drone6[6307]: /go/src/github.com/drone/drone/vendor/github.com/gin-gonic/gin/context.go:97 (0x8baa1a)
Sep  5 21:46:12 drone6-server drone6[6307]: /go/src/github.com/drone/drone/router/middleware/store.go:15 (0x9cd767)
Sep  5 21:46:12 drone6-server drone6[6307]: /go/src/github.com/drone/drone/vendor/github.com/gin-gonic/gin/context.go:97 (0x8baa1a)
Sep  5 21:46:12 drone6-server drone6[6307]: /go/src/github.com/drone/drone/vendor/github.com/gin-gonic/contrib/ginrus/ginrus.go:26 (0x9cdbbf)
Sep  5 21:46:12 drone6-server drone6[6307]: /go/src/github.com/drone/drone/vendor/github.com/gin-gonic/gin/context.go:97 (0x8baa1a)
Sep  5 21:46:12 drone6-server drone6[6307]: /go/src/github.com/drone/drone/router/middleware/header/header.go:25 (0x8cb040)
Sep  5 21:46:12 drone6-server drone6[6307]: /go/src/github.com/drone/drone/vendor/github.com/gin-gonic/gin/context.go:97 (0x8baa1a)
Sep  5 21:46:12 drone6-server drone6[6307]: /go/src/github.com/drone/drone/router/middleware/header/header.go:17 (0x8cafe9)
Sep  5 21:46:12 drone6-server drone6[6307]: /go/src/github.com/drone/drone/vendor/github.com/gin-gonic/gin/context.go:97 (0x8baa1a)
Sep  5 21:46:12 drone6-server drone6[6307]: /go/src/github.com/drone/drone/vendor/github.com/gin-gonic/gin/recovery.go:45 (0x8c8caa)
Sep  5 21:46:12 drone6-server drone6[6307]: /go/src/github.com/drone/drone/vendor/github.com/gin-gonic/gin/context.go:97 (0x8baa1a)
Sep  5 21:46:12 drone6-server drone6[6307]: /go/src/github.com/drone/drone/vendor/github.com/gin-gonic/gin/gin.go:284 (0x8c0850)
Sep  5 21:46:12 drone6-server drone6[6307]: /go/src/github.com/drone/drone/vendor/github.com/gin-gonic/gin/gin.go:265 (0x8c016b)
Sep  5 21:46:12 drone6-server drone6[6307]: /usr/local/go/src/net/http/server.go:2568 (0x6ceb92)
Sep  5 21:46:12 drone6-server drone6[6307]: /usr/local/go/src/net/http/server.go:1825 (0x6cad32)
Sep  5 21:46:12 drone6-server drone6[6307]: /usr/local/go/src/runtime/asm_amd64.s:2197 (0x45dda1)
```

**Comment by bradrydzewski** [comment]
thanks, I'll be sure to include a fix in 0.8-rc.4

**Comment by bradrydzewski** [comment]
ok, I have some unit tests around this, and feel confident making the claim it is closed in 0.8.0-rc.5
